src/modules/filter-drawer/filter-drawer.tsx

```diff
@@ -0,0 +1,349 @@
+
+                    {shouldShowClearFiltersButton && (
+                      <Header.Action testID="drawer-clear-filters-action">
+                        <Link
+                          onPress={onClearFilters}
+                          label="Clear Filters"
+                          small
+                        />
+                      </Header.Action>
+                    )}
+                  </Animated.View>
+                </Header.Wrapper>
+
+                {primaryChildren}
+              </ScrollView>
+
+              {/* Footer */}
+              <FooterWrapper>
+                <SafeAreaView>
+                  <ButtonGroup>
+                    {shouldShowCancelButton && (
+                      <Button
+                        color="secondary"
+                        appearance="outline"
+                        testID="drawer-cancel-filters-button"
+                        onPress={onCancel}>
+                        Cancel
+                      </Button>
+                    )}
+
+                    <Button
+                      color="success"
+                      testID="drawer-apply-filters-button"
+                      onPress={onApplyFilters}>
+                      Apply Filters
+                    </Button>
+                  </ButtonGroup>
+                </SafeAreaView>
+              </FooterWrapper>
+            </PrimaryContentWrapper>
+
+            {/* Secondary Content */}
+            <SecondaryContentWrapper testID="drawer-secondary-content">
+              <ScrollView showsVerticalScrollIndicator={false}>
+                {/* Header */}
+                <Header.Wrapper>
+                  <Animated.View
+                    style={{
+                      opacity: headerBackOpacity,
+                      transform: [{translateX: headerBackTranslateX}],
+                    }}>
+                    <Header.BackAction testID="drawer-back-action">
+                      <Link onPress={onBackToPrimaryContent}>
+                        <Icon file={IconChevronLeft} />
+                      </Link>
+
+                      {secondaryViewTitle ? (
+                        <Header.BackActionTitle>
+                          <Heading variant="h2">{secondaryViewTitle}</Heading>
+                        </Header.BackActionTitle>
+                      ) : null}
+                    </Header.BackAction>
+                  </Animated.View>
+                </Header.Wrapper>
+
+                {secondaryChildren}
+              </ScrollView>
+            </SecondaryContentWrapper>
+          </AnimatedWrapper>
+        </SectionWrapper>
+
+        {/* Close Button */}
+        <AnimatedCloseWrapper style={{opacity: closeButtonOpacity}}>
+          <TouchableOpacity
+            testID="drawer-close-button"
+            disabled={shouldShowSecondaryContent}
+            onPress={onCloseModal}>
+            <Icon file={IconCloseXCircle} />
```

Comment:
please increase the size of close icon upto 32px. and also make top and right margins to 11px.